### ISSUE TITLE
'Segmentation fault (core dumped)' trying to change parameter in the middle of training

### ISSUE THREAD

**Issue by Ubikas** [Microsoft/LightGBM#865]
## Environment info
Operating System: Ubuntu 16.04
CPU: Intel® Core™ i7-7700K CPU @ 4.20GHz × 8
GPU: GeForce GTX 1080/PCIe/SSE2 (Driver Version: 384.59; cuda_8.0.61.2 [patched])
C++/Python/R version: gcc 5.4/ py 2.7
LightGBM v. 2.0.5 (just compiled with GPU option from the branch two weeks ago)

## Error Message: 
Segmentation fault (core dumped)

## Reproducible examples
https://github.com/Microsoft/LightGBM/blob/master/examples/python-guide/advanced_example.py
line 101
## Steps to reproduce
run advanced_example.py

##Additional info
It crashes only when the time comes to change a parameter. This is info from my model, but it crashes on advanced_example.py too while trying to change learning_rate at line 101:

<...>
Finish first 10 rounds...
('7th feature name is:', "'feature_6'")
Load model to predict
("The rmse of loaded model's prediction is:", 0.46292456076369248)
[LightGBM] [Info] Finished loading 10 models
("The rmse of pickled model's prediction is:", 0.47115339195015815)
[LightGBM] [Info] Warning: last line of model.txt has no end of line, still using this line
[LightGBM] [Info] Finished loading 10 models
/usr/local/lib/python2.7/dist-packages/lightgbm/basic.py:986: UserWarning: Using categorical_feature in Dataset.
  warnings.warn('Using categorical_feature in Dataset.')
[LightGBM] [Info] Trained a tree with leaves=31 and max_depth=8
[11]	valid_0's binary_logloss: 0.616177
[LightGBM] [Info] Trained a tree with leaves=31 and max_depth=8
[12]	valid_0's binary_logloss: 0.611792
[LightGBM] [Info] Trained a tree with leaves=31 and max_depth=8
[13]	valid_0's binary_logloss: 0.607043
[LightGBM] [Info] Trained a tree with leaves=31 and max_depth=7
[14]	valid_0's binary_logloss: 0.602556
[LightGBM] [Info] Trained a tree with leaves=31 and max_depth=8
[15]	valid_0's binary_logloss: 0.598251
[LightGBM] [Info] Trained a tree with leaves=31 and max_depth=9
[16]	valid_0's binary_logloss: 0.595061
[LightGBM] [Info] Trained a tree with leaves=31 and max_depth=9
[17]	valid_0's binary_logloss: 0.591851
[LightGBM] [Info] Trained a tree with leaves=31 and max_depth=8
[18]	valid_0's binary_logloss: 0.588475
[LightGBM] [Info] Trained a tree with leaves=31 and max_depth=8
[19]	valid_0's binary_logloss: 0.585863
[LightGBM] [Info] Trained a tree with leaves=31 and max_depth=8
[20]	valid_0's binary_logloss: 0.583035
[LightGBM] [Info] Finished loading 20 models
Finish 10 - 20 rounds with model file...
[LightGBM] [Info] Trained a tree with leaves=31 and max_depth=7
[21]	valid_0's binary_logloss: 0.58106
Segmentation fault (core dumped)

I get the same error in my model using callbacks lgb.cv: callbacks=[lgb.reset_parameter(learning_rate=[0.05]*200 + [0.045]*100 + [0.04]*700)]
Actually it crashes on trying to change any parameter, it is not learning_rate only. With verbose:10 I get

[LightGBM] [Info] Trained a tree with leaves=44 and max_depth=8
[LightGBM] [Info] No further splits with positive gain, best gain: -inf
[LightGBM] [Info] Trained a tree with leaves=26 and max_depth=8
[100]	cv_agg's binary_logloss: 0.0921139
[LightGBM] [Debug] use subset for bagging
[LightGBM] [Debug] use subset for bagging
[LightGBM] [Debug] use subset for bagging
[LightGBM] [Debug] use subset for bagging
[LightGBM] [Debug] use subset for bagging
Segmentation fault (core dumped)

I don't know why it mentions bagging in this case, because the parameter i'm changing is learning_rate. But I do use bagging parameters:

'bagging_fraction': 0.95, 
'bagging_freq': 40,
'bagging_seed': 0,

What can I do to make it work? Changing learning_rate in the training saves time and increase accuracy.
Thanks.

**Comment by guolinke** [comment]
@wxchan
